src/main/scala/strawman/collection/Factories.scala

```diff
@@ -22,35 +24,62 @@ trait FromSpecificIterable[-A, +C] extends Any {
   def fromSpecificIterable(it: Iterable[A]): C
 }
 
+/**
+  * Builder factory
+  * @tparam A Element type (e.g. `Int`)
+  * @tparam C Collection type (e.g. `List[Int]`)
+  */
+trait CanBuild[-A, C] {
+  /** Creates a builder */
+  def newBuilder(): Builder[A, C]
+}
+
 /** Base trait for companion objects of unconstrained collection types */
-trait IterableFactory[+CC[_]] extends FromIterable[CC] {
+trait IterableFactory[CC[_]] extends FromIterable[CC] {
   def empty[A]: CC[A]
 
   def apply[A](xs: A*): CC[A] = fromIterable(View.Elems(xs: _*))
 
   def fill[A](n: Int)(elem: => A): CC[A] = fromIterable(View.Fill(n)(elem))
+
+  def newBuilder[A](): Builder[A, CC[A]]
+
+  implicit def canBuild[A]: CanBuild[A, CC[A]] = () => newBuilder[A]()
```

Comment:
I think this is problematic. Not all collections admit a builder.E.g. LazyList does not. Also, I am 
not happy that every factory now has two ways to construct things, `fromIterable` and `newBuilder`. What argument is there that we are not getting back into the complexity we just struggled to get out of?